students/stuartkershaw/lesson07/html_render.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+
+
+class TextWrapper:
+    """
+    A simple wrapper that creates a class with a render method
+    for simple text
+    """
+    def __init__(self, text):
+        self.text = text
+
+    def render(self, file_out, cur_ind=""):
+        file_out.write(cur_ind)
+        file_out.write(
+            f"{self.text}"
+        )
+
+
+class Element:
+    tag_name = ''
+    indentation = 4
+    cur_tree_level = [0]
+    ignored_tag_spacing = ['', 'a', 'html']
+
+    def __init__(self, content=None, **kwargs):
+        self.content = [] if content is None else [TextWrapper(content)]
+        self.kwargs = kwargs
+
+    def append(self, content):
+        if isinstance(content, str):
+            self.content.append(TextWrapper(content))
+        else:
+            self.content.append(content)
+
+    def generate_spacing(self, tree_level):
+        self.tree_level = tree_level
+
+        return " " * self.tree_level * self.indentation
+
+    def apply_attributes(self, attrs):
+        self.attrs = attrs
+
+        for key, value in self.attrs.items():
+            self.file_out.write(f" {key}=\"{value}\"")
+
+    def render(self, file_out, cur_ind=""):
+        self.file_out = file_out
```

Comment:
there is usually not a good reason to create an instance attribute outside of `__init__` which is why linters will usually complain about that

Reply:
Hi @natasha-aleksandrova - I'm underway with patching this up, but I'm having some difficulty imagining how to change this. With the way that `run_html_render.py` passes `file_out` to the render method, I'm not understanding how to shift that to `__init__` without changing `run_html_render.py`... am I just overlooking something there? Thanks -